Implement IsSampled for OpenTelemetry bridgeSpanContext (#3570)
open-telemetry/opentelemetry-go: bridge/opentracing/README.md

	// Propagate the otSpan to both OpenTracing and OpenTelemetry
	// instrumentation by using the ctxWithOTAndOTelSpan context.
```

## Extended Functionality

> While the bridge does not expose functionality that is not implemented by OpenTelemetry, it does expose some that is part of OpenTelemetry API and not OpenTracing API.

Reviewer: ```suggestion
The bridge functionality can be extended beyond the OpenTracing API.
```

> **`SpanContext.IsSampled`**

Reviewer: ```suggestion
### `SpanContext.IsSampled`
```

> Proxies underlying `trace.IsSampled` method (see [documentation](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext.IsSampled)). In order to use it, you have to cast it:

Reviewer: ```suggestion
Return the underlying OpenTelemetry [`Span.IsSampled`](https://pkg.go.dev/go.opentelemetry.io/otel/trace#SpanContext.IsSampled) value by converting a `bridgeSpanContext`.
```

```go
type samplable interface {
	IsSampled() bool
}

var sc opentracing.SpanContext = ...

> if sc.(samplable).IsSampled() {
> 	// Span is expected to be sampled.
> } else {
> 	// Span will be discarded.
> }

Reviewer: ```suggestion
if s, ok := sc.(samplable); ok && s.IsSampled() {
	// Do something with sc knowing it is sampled.
}
```
